client/my-sites/domains/domain-management/transfer/transfer-out/unlocked.jsx

```diff
@@ -11,28 +12,65 @@ import Card from 'components/card';
 import SectionHeader from 'components/section-header';
 import { getSelectedDomain } from 'lib/domains';
 import Button from 'components/button';
-import { requestTransferCode, enableDomainLocking } from 'lib/upgrades/actions';
+import { requestTransferCode, cancelTransferRequest } from 'lib/upgrades/actions';
 import notices from 'notices';
 import { displayRequestTransferCodeResponseNotice } from './shared';
 import support from 'lib/url/support';
 
 class Unlocked extends React.Component {
-	state = {
-		submitting: false
-	};
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			submitting: false,
+			// For protected domains, the flow in transfer-out/index will
+			// send users to transfer-out/locked and send an initial auth
+			// code request from there before the user gets here.
+			sent: ! this.isDomainAlwaysTransferrable(),
+		};
+	}
+
+	componentWillUnmount() {
+		this.setStateIfMounted = noop;
+	}
+
+	/**
+	 * Wrap setState calls that might occur after unmounting.
+	 *
+	 * When we cancel a transfer, that might update locking or privacy,
+	 * but errors mean we can't know in time - the store gets the information
+	 * before we do.
+	 *
+	 * The recommended solution is cancellable promises, but we don't want to
+	 * cancel these requests if we navigate away, so that won't work for us here.
+	 */
+	setStateIfMounted( ...args ) {
```

Comment:
It occurs to me that if this pattern is generally acceptable we could pull it out into an higher order component.